community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/CypherPreParser.scala

```diff
@@ -46,8 +46,8 @@ case object CypherPreParser extends Parser with Base {
 
   def PlannerOption: Rule1[PreParserOption] = rule("planner option") (
       option("planner", "cost") ~ push(CostPlannerOption)
-    | option("planner", "greedy") ~ push(GreedyPlannerOption)
     | option("planner", "rule") ~ push(RulePlannerOption)
+    | option("planner", "greedy") ~ push(GreedyPlannerOption)
```

Comment:
In Neo4j 3.0, it's still OK to write:
```CYPHER 2.3 planner=greedy <query>```

That's why the concept is still lingering here.